### PR TITLE
drivers: serial: nrfx_uarte: Add support for ENDTX_STOPTX short

### PR DESCRIPTION
**Pull request by nordic-krch**
nrf54x has new feature: ENDTX_STOPTX short. Add devicetree property to UARTE. Use short is property is set.

### CODE REVIEW

**Comment by pdgendt** [comment]
Remove the clang format commit, the messages are informative, but formatting an entire file will mess up the history which we don't want, I think.

**Review by bjarki-andreasen — DISMISSED** [review]
I really like the flag in the `short-endtx-stoptx` dt prop :)

File: dts/bindings/serial/nordic,nrf-uarte.yaml [hunk 1]
```diff
@@ -3,3 +3,9 @@ description: Nordic nRF family UARTE (UART with EasyDMA)
 compatible: "nordic,nrf-uarte"
 
 include: ["nordic,nrf-uart-common.yaml", "memory-region.yaml"]
+
+properties:
+  short-endtx-stoptx:
```
[the remark this answers is not in the document]
Comment (nordic-krch) [reply dts/bindings/serial/nordic,nrf-uarte.yaml]: renamed to `endtx-stoptx-supported`

File: drivers/serial/uart_nrfx_uarte.c [hunk 2]
```diff
@@ -85,6 +89,12 @@ LOG_MODULE_REGISTER(uart_nrfx_uarte, CONFIG_UART_LOG_LEVEL);
 #define UARTE_ENHANCED_POLL_OUT 1
 #endif
 
+#define INSTANCE_PROP(unused, prefix, i, prop) UARTE_PROP(prefix##i, prop)
+
+#if UARTE_FOR_EACH_INSTANCE(INSTANCE_PROP, (+), (0), short_endtx_stoptx)
+#define UARTE_HAS_ENDTX_STOPTX_SHORT 1
+#endif
```
[the remark this answers is not in the document]
Comment (nordic-krch) [reply drivers/serial/uart_nrfx_uarte.c]: added comparison against total number of enabled instances and only then short is used.

File: dts/common/nordic/nrf54h20.dtsi [hunk 3]
```diff
@@ -625,6 +625,7 @@
 				status = "disabled";
 				interrupts = <230 NRF_DEFAULT_IRQ_PRIORITY>;
 				clocks = <&hsfll120>;
+				endtx-stoptx-supported;
```
[the remark this answers is not in the document]
Comment (nordic-krch) [reply dts/common/nordic/nrf54h20.dtsi:628]: done

[the remark this answers is not in the document]
Comment (anangl) [reply dts/common/nordic/nrf54h20.dtsi:628]: > done

nRF9280 has more than one UART :)

**Review by anangl — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Pull request merged**